src/main/java/org/openstreetmap/atlas/geography/sharding/DynamicTileSharding.java

```diff
@@ -114,6 +118,49 @@ public boolean equals(final Object other)
             return false;
         }
 
+        /**
+         * Does a deep equals with the other node
+         * 
+         * @param other
+         *            other Node
+         * @return true if entire structure is equal, false if not
+         */
+        private boolean deepEquals(final Node other)
+        {
+            final Comparator<Node> nodeCompare = Comparator.comparing(Node::getTile);
+            // BFS through both trees to get equality
+            final Queue<Node> queue = new LinkedList<>();
+            queue.offer(this);
+            queue.offer(other);
+            while (!queue.isEmpty())
+            {
+                final Node node1 = queue.poll();
+                final Node node2 = queue.poll();
```

Comment:
Maybe I'm not reading this correctly, you're checking that the `queue` is not empty before polling and then polling twice. Is there a chance that the queue only has 1 element and when you poll the second time, `node2` is `null` and a NP happens on line 139 when `node2.getChildren()` is called?

Reply:
Initially I did this, https://github.com/osmlab/atlas/pull/441/commits/3beac046d3331c5a90ee2ef6c20ab6254d29c742) but then sonar complained I wasn't checking for the existence of the first node. At which point I realized I didn't need to worry about it because I check the size of the children to make sure they're equal. So I am always adding two nodes at a time.

Reply:
Less safe if someone changes the function later. But, removes unnecessary checking.

Reply:
That makes perfect sense, thanks for the explanation! Maybe worth a comment in case the next reader has the same question.